Create a price feed for gOHM

tonypiano opened the pull request:
This combines two price feeds the ETH-USD feed and a OHM-ETH feed that
is currently supported by Chainlink. This allows us to reuse all the
existing smarts for detecting bad oracles and falling back to Tellor.

Commit:
Create a price feed for gOHM

This combines two price feeds the ETH-USD feed and a OHM-ETH feed that
is currently supported by Chainlink. This allows us to reuse all the
existing smarts for detecting bad oracles and falling back to Tellor.

Commit:
Add a couple more test cases

kushal256 commented on packages/contracts/contracts/MultiPriceFeed.sol:
```diff
@@ -0,0 +1,49 @@
+import "./Dependencies/Ownable.sol";
+import "./Dependencies/CheckContract.sol";
+
+contract MultiPriceFeed is Ownable, CheckContract, IPriceFeed {
```
Really like how you made this implement IPriceFeed ;)

kushal256 commented on packages/contracts/contracts/MultiPriceFeed.sol:
```diff
@@ -0,0 +1,49 @@
+        uint256 ohmETHPrice = ohmETHFeed.fetchPrice();
+        uint256 ethUSDPrice = ethUSDFeed.fetchPrice();
+
+        require(ohmETHPrice > 0, "PriceFeed: OHM-ETH price is zero");
```
Do we need to check `PriceFeed::status()` ?  

It might make sense to return the last good price if either underlying feed has `status != Status.chainlinkWorking`, as this mimics the behavior of the underlying `PriceFeed::fetchPrice()`

If we did that might want to also keep track of the aggregated status within here, as well.

What do you think?

tonypiano replied:
My idea here was to let the underlying contracts handle their statuses. `fetchPrice` will always return a result. The multi-price feed then doesn't need to care if the OHM-ETH feed is using Tellor and the ETH-USD feed is using Chainlink. These requires are just here to be extra cautious in case there is a bug in their `PriceFeed` code that will liquidate everybody.

kushal256 replied:
Was trying to think thru this scenario:

`ETH-USD = Status.bothOraclesUntrusted`
`OHM-ETH = Status.chainlinkWorking`

In that scenario, we'd take the `PriceFeed::lastGoodPrice` from `ETH-USD`, but use the current price for `OHM-ETH`.

Is that better or worse than just taking the `lastGoodPrice` from each feed?  I'm not really sure as I don't know how old each respective `lastGoodPrice` is, but it seems like it would be less risky than using both an old and a current price? 

I wonder how often both Chainlink and tellor price feeds both go down, maybe I'm overthinking a scenario that won't occur

tonypiano replied:
Yeah, I don't really know what the best option is. I wonder if we can see how the ETH-USD feed holds up in liquity. All this complexity could be for nothing. I'll try to dig in

kushal256 replied:
Just added some logic to account for these scenarios.  Think the complexity is minor, and makes it safer.  Also added some events

tonypiano replied:
I love it. Great stuff

kushal256 commented on packages/contracts/test/MultiPriceFeedTest.js:
```diff
@@ -0,0 +1,99 @@
+const testHelpers = require("../utils/testHelpers.js")
+const {dec, toBN} = testHelpers.TestHelper;
+
+contract('PriceFeed', async accounts => {
```
Loving these test cases!

Commit:
-MutliPriceFeed: add checks to underlying price feed status to determine if we should sent last good price or not.  added events MultiPriceFeedStatusChanged and LastGoodMultiPriceUpdated
-Moved PriceFeed.Status enum up to IPriceFeed for re-use within MultiPriceFeed
-Add test cases

Commit:
cleanup

Commit:
better tests
